### PR TITLE
Feature/add multiple identical key handling for inventory

### PR DESCRIPTION
**Pull request by seilc1**
The inventory service didn't handle multiple service details with the same key 'properly'. I rectified this with adding a new function getAll and replaceByKey.

### CODE REVIEW

**Comment by seilc1** [comment]
> I'd like to have parameter validation for the public methods, I added a commit with an example test showing the expectations.
> 
> Other than that, the method `get(String serviceInstanceGuid)` might be a little confusing next to `getAll(String serviceInstanceGuid, String key)`. Those could potentially be replaced with `getAllByServiceInstanceGuid(String serviceInstanceGuid)` and `getAllByKey(String serviceInstanceGuid, String key)`. In that case I'd suggest to deprecate the `get(String serviceInstanceGuid)`.

I've added documentation to the interface to clarify the function of the functions ;-) I think the name in combination with the parameters is already saying enough.

File: broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/inventory/LocalInventoryServiceImpl.groovy [hunk 1]
```diff
@@ -25,41 +36,59 @@ class LocalInventoryServiceImpl implements InventoryService {
...
     }
 
     @Override
     Pair<String, String> get(String serviceInstanceGuid, String key, String defaultValue) {
-        def detail = getServiceInstance(serviceInstanceGuid).details.find { d -> d.key.equalsIgnoreCase(key) }
+        checkArgument(isNotBlank(serviceInstanceGuid), ERROR_SERVICE_INSTANCE_ID_NOT_DEFINED)
+        checkArgument(isNotBlank(key), ERROR_KEY_NOT_DEFINED)
+
+        def details = getServiceInstance(serviceInstanceGuid).details.findAll { d -> d.key.equalsIgnoreCase(key) }
+        checkState(details.size() <= 1, ERROR_DETAIL_NOT_UNIQUE, key)
```
Comment (ampersand8) [inline broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/inventory/LocalInventoryServiceImpl.groovy]: Why is it not unique when `size() <= 1`?

File: broker/core/src/test/groovy/com/swisscom/cloud/sb/broker/services/inventory/LocalInventoryServiceImplSpec.groovy [hunk 2]
```diff
@@ -132,7 +128,7 @@ class LocalInventoryServiceImplSpec extends Specification {
         result.second == "default_value_001"
```
Comment (ampersand8) [inline broker/core/src/test/groovy/com/swisscom/cloud/sb/broker/services/inventory/LocalInventoryServiceImplSpec.groovy]: I'd for sure also put the behaviour, that the default value is added to the result even when a key is found, to the interface doc.

File: broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/inventory/LocalInventoryServiceImpl.groovy [hunk 3]
```diff
@@ -25,41 +36,59 @@ class LocalInventoryServiceImpl implements InventoryService {
...
-        def detail = getServiceInstance(serviceInstanceGuid).details.find { d -> d.key.equalsIgnoreCase(key) }
+        checkArgument(isNotBlank(serviceInstanceGuid), ERROR_SERVICE_INSTANCE_ID_NOT_DEFINED)
+        checkArgument(isNotBlank(key), ERROR_KEY_NOT_DEFINED)
 
-        if (detail == null) {
-            ErrorCode.SERVICEBROKERSERVICEPROVIDER_INTERNAL_SERVER_ERROR.throwNew("No serviceDetail with key:${key} found.")
-        }
+        def details = getServiceInstance(serviceInstanceGuid).details.findAll { d -> d.key.equalsIgnoreCase(key) }
+        checkState(details.size() > 0, ERROR_DETAIL_NOT_FOUND, key)
+        checkState(details.size() == 1, ERROR_DETAIL_NOT_UNIQUE, key)
```
[the remark this answers is not in the document]
Comment (seilc1) [reply broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/inventory/LocalInventoryServiceImpl.groovy]: it checks that the size() is == 1 ... so the exception is only thrown if it is != 1 - preconditions are sometimes confusing

File: broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/inventory/LocalInventoryServiceImpl.groovy (hunk 1 above)
Comment (seilc1) [reply broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/inventory/LocalInventoryServiceImpl.groovy]: same as above

File: broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/inventory/LocalInventoryServiceImpl.groovy (hunk 3 above)
[the remark this answers is not in the document]
Comment (ampersand8) [reply broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/services/inventory/LocalInventoryServiceImpl.groovy]: Of course, yeah got confused.

**Pull request merged**